### ISSUE TITLE
Terrible bug in DoubleArrayTrie#resolveConflict()

### ISSUE THREAD

**Issue by ghost** [digitalstain/DoubleArrayTrie#3]
I am using your DoubleArrayTrie project for some indexing project of mine, and I discovered a terrible bug: method resolveConflict(), line 

if ( tempNext < getSize() && getCheck(tempNext) == s)

should read: 

if ( tempNext >= 0 && tempNext < getSize() && getCheck(tempNext) == s)

as tempNext , which is equal to getBase(s) + c, may be equal to -2, which throws an ArrayIndexOutofBoundsException, or an AssertionError if you are so lucky as to be running with -ea. I.e., in case of resolving a conflict for a leaf node, you'll always run through this faulty implementation.

**Comment by digitalstain** [comment]
@exercitussolus Thanks for your report. This code hasn't been maintained in a while so it is expected that some pretty serious bugs will be present, especially since the context i used it in originally was quite narrow in it's use. However, i'll try to make some time and take care of this.

Do you happen to have a test case demonstrating the problem?

**Comment by ghost** [comment]
Yes, I do, although I can not upload the actual file I use here ( it is a large web crawler log file ). 
It consists of 10,000 ( ten thousand ) lines of this format: 

2012-09-24T02:00:53.520Z   302          0 http://www.toesocks.co.uk/catalog/product_compare/remove/product/261/uenc/aHR0cDovL3d3dy50b2Vzb2Nrcy5jby51ay9vdXRkb29ycy5odG1sP21hdGVyaWFsPTY1/ LLL http://www.toesocks.co.uk/outdoors.html?material=65 text/html #110 20120924020052933+585 sha1:3I42H3S6NNFQ2MSVX7XZKYAYSCX5QBYJ - 184.173.24.67

2012-09-24T02:00:53.846Z   200       6520 http://thumbs3.ebaystatic.com/pict/3007685281104040_1.jpg LLLE http://www.an0n.co.uk/buy/garmin+gps+nuvi+1450 image/jpeg #153 20120924020053474+367 sha1:IBCVFRZ3VT5NIVCFNNUNQRJG7BYTLSYA - 195.245.125.163

2012-09-24T02:00:56.126Z   200       2582 http://thumbs4.ebaystatic.com/pict/2607314860594040_2.jpg LPLLE http://www.digico1.co.uk/Home-Audio-Hi-Fi/Other-Home-Audio image/jpeg #181 20120924020055675+448 sha1:AVOTD2K4L3BQRCF3V6RML2HPDL3XPI55 - 195.245.125.187

2012-09-24T02:00:56.253Z   301          0 http://shop.ebay.co.uk/i.html?_nkw=saints+row+xbox+360 LLLLLLL http://keyword-index.ebay.co.uk/L1-3-426.html unknown #043 20120924020055709+542 sha1:3I42H3S6NNFQ2MSVX7XZKYAYSCX5QBYJ - 66.135.203.102

2012-09-24T02:00:57.337Z   302          0 http://www.myhillspet.co.uk/catalog/product_compare/remove/product/5867/uenc/aHR0cDovL3d3dy5teWhpbGxzcGV0LmNvLnVrL2ZlbGluZS9zY2llbmNlLXBsYW4vb3B0aW1hbC1jYXJlL2YtYWQtc2FsbW9uMjR4ODVnLWNhbnMuaHRtbA,,/ LLL http://www.myhillspet.co.uk/feline/science-plan/optimal-care/f-ad-salmon24x85g-cans.html text/html #036 20120924020057170+165 sha1:3I42H3S6NNFQ2MSVX7XZKYAYSCX5QBYJ - 109.123.80.47

What I did: 
- wrote a ByteNaturalMapping, implementing your NaturalMapping interface
- read this file in byte[ ] blocks ( block size is irrelevant )
- tokenized the byte[ ] blocks into byte[ ][ ] tokens, filtering upon white spaces and CR / CRLF
- mapped each byte[ ][ ] block to an IntegerArrayList with ByteNaturalMapping
- wit the resulting IntegerArrayLists, did an addToTrie( IntegerArrayList )

What happens then is an IndexOutOfBoundsException: -2

thrown by the resolveConflict method in line 149

My proposed fix solves this bug.
